dandi/dandiapi.py

```diff
@@ -0,0 +1,348 @@
+        else:
+            return getattr(requests, method.lower())
+
+    def send_request(
+        self,
+        method,
+        path,
+        parameters=None,
+        data=None,
+        files=None,
+        json=None,
+        headers=None,
+        json_resp=True,
+        **kwargs,
+    ):
+        """
+        This method looks up the appropriate method, constructs a request URL
+        from the base URL, path, and parameters, and then sends the request. If
+        the method is unknown or if the path is not found, an exception is
+        raised, otherwise a JSON object is returned with the response.
+
+        This is a convenience method to use when making basic requests that do
+        not involve multipart file data that might need to be specially encoded
+        or handled differently.
+
+        :param method: The HTTP method to use in the request (GET, POST, etc.)
+        :type method: str
+        :param path: A string containing the path elements for this request.
+            Note that the path string should not begin or end with the path  separator, '/'.
+        :type path: str
+        :param parameters: A dictionary mapping strings to strings, to be used
+            as the key/value pairs in the request parameters.
+        :type parameters: dict
+        :param data: A dictionary, bytes or file-like object to send in the body.
+        :param files: A dictionary of 'name' => file-like-objects for multipart encoding upload.
+        :type files: dict
+        :param json: A JSON object to send in the request body.
+        :type json: dict
+        :param headers: If present, a dictionary of headers to encode in the request.
+        :type headers: dict
+        :param json_resp: Whether the response should be parsed as JSON. If False, the raw
+            response object is returned. To get the raw binary content of the response,
+            use the ``content`` attribute of the return value, e.g.
+
+            .. code-block:: python
+
+                resp = client.get('my/endpoint', json_resp=False)
+                print(resp.content)  # Raw binary content
+                print(resp.headers)  # Dict of headers
+
+        :type json_resp: bool
+        """
+        if not parameters:
+            parameters = {}
+
+        # Look up the HTTP method we need
+        f = self._request_func(method)
+
+        url = self.get_url(path)
+
+        # Make the request, passing parameters and authentication info
+        _headers = headers or {}
+
+        if json_resp and "accept" not in _headers:
+            _headers["accept"] = "application/json"
+
+        result = f(
+            url,
+            params=parameters,
+            data=data,
+            files=files,
+            json=json,
+            headers=_headers,
+            **kwargs,
+        )
+
+        # If success, return the json object. Otherwise throw an exception.
+        if result.status_code not in (200, 201):
```

Comment:
I suggest using "`if not result.ok`" instead, as that only regards 4xx and 5xx status codes as errors.  There are other 2xx codes that aren't errors, plus the 3xx codes.

Reply:
THANKS! done in a45b550951fefa89e847052a1eb919fbb480758e